app/javascript/components/bookmarks/BookmarkManager.jsx

```diff
@@ -0,0 +1,286 @@
+import { Popover, OverlayTrigger } from 'react-bootstrap'
+import BookmarksList from '~/components/bookmarks/BookmarksList'
+import { fetchBookmarks, createBookmark, updateBookmark, deleteBookmark } from '~/lib/scp-api'
+import React, { useEffect, useState } from 'react'
+import _cloneDeep from 'lodash/clone'
+import { isUserLoggedIn } from '~/providers/UserProvider'
+import { useLocation } from '@reach/router'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faLink, faUndo } from '@fortawesome/free-solid-svg-icons'
+import useErrorMessage from '~/lib/error-message'
+
+/**
+ * Component to bookmark views in Explore tab, as well as existing link/reset buttons
+ *
+ * @param bookmarks {Array} existing user bookmark objects
+ * @param clearExploreParams
+ */
+export default function BookmarkManager({bookmarks, clearExploreParams}) {
+  const location = useLocation()
+  const [allBookmarks, setAllBookmarks] = useState(bookmarks)
+  const [saveText, setSaveText] = useState('Save')
+  const [saveDisabled, setSaveDisabled] = useState(false)
+  const [bookmarkSaved, setBookmarkSaved] = useState(null)
+  const [deleteDisabled, setDeleteDisabled] = useState(false)
+  const { ErrorComponent, setShowError, setError } = useErrorMessage()
+  const DEFAULT_BOOKMARK = {
+    name: '',
+    description: '',
+    path: getBookmarkPath()
+  }
+
+  /** copies the url to the clipboard */
+  function copyLink() {
+    navigator.clipboard.writeText(location.href)
+  }
+
+  /** Find a matching bookmark using the current URL params, or return default empty bookmark  */
+  function findExistingBookmark() {
+    const bookmark = allBookmarks.find(bookmark => { return bookmark.path === getBookmarkPath() })
+    if (bookmark) {
+      setBookmarkSaved(true)
+      return bookmark
+    } else {
+      setBookmarkSaved(false)
+      return DEFAULT_BOOKMARK
+    }
+  }
+
+  const [currentBookmark, setCurrentBookmark] = useState(findExistingBookmark)
+
+  /** get bookmark ID from BSON object */
+  function getBookmarkId(bookmark) {
+    return bookmark?._id || bookmark?.id
+  }
+
+  /** Add a bookmark to the list of user bookmarks, or update ref of existing */
+  function updateBookmarkList(bookmark) {
+    setBookmarkSaved(true)
+    const existingIdx = allBookmarks.findIndex(bookmark => { return bookmark.path === getBookmarkPath() })
+    if (existingIdx >= 0) {
+      setAllBookmarks(prevBookmarks => {
+        prevBookmarks[existingIdx] = bookmark
+        return prevBookmarks
+      })
+    } else {
+      setAllBookmarks(prevBookmarks => [...prevBookmarks, bookmark])
+    }
+    setServerBookmarksLoaded(false)
+    setCurrentBookmark(bookmark)
+  }
+
+  /** remove a deleted bookmark from the list */
+  function removeBookmarkFromList(bookmarkId) {
+    const remainingBookmarks = allBookmarks.filter(bookmark => {
+      return getBookmarkId(bookmark) !== bookmarkId
+    })
```

Comment:
More conventional, simpler:

```suggestion
    const remainingBookmarks = allBookmarks.filter(
      bookmark => getBookmarkId(bookmark) !== bookmarkId
    )
```